sdk/include/string.h

```diff
@@ -6,27 +6,37 @@
 #include <stddef.h>
 #include <stdint.h>
 
-int __cheri_libcall    memcmp(const void *str1,
-                              const void *str2,
-                              size_t      count) __asm__("memcmp");
-void *__cheri_libcall  memcpy(void       *dest,
-                              const void *src,
-                              size_t      n) __asm__("memcpy");
-void *__cheri_libcall  memset(void *, int, size_t) __asm__("memset");
-void *__cheri_libcall  memmove(void       *dest,
-                               const void *src,
-                               size_t      n) __asm__("memmove");
-void *__cheri_libcall  memchr(const void *, int, size_t);
-void *__cheri_libcall  memrchr(const void *, int, size_t);
-size_t __cheri_libcall strlen(const char *str);
-int __cheri_libcall    strncmp(const char *s1, const char *s2, size_t n);
-char *__cheri_libcall  strncpy(char *dest, const char *src, size_t n);
-int __cheri_libcall    strcmp(const char *s1, const char *s2);
-char *__cheri_libcall  strnstr(const char *haystack,
-                               const char *needle,
-                               size_t      haystackLength);
-char *__cheri_libcall  strchr(const char *s, int c);
-size_t __cheri_libcall strlcpy(char *dest, const char *src, size_t n);
+#define DECLARE_STRING_LIBCALL(rtype, name, ...)                               \
+	rtype __cheri_libcall name(__VA_ARGS__) __asm__(#name);
+
+DECLARE_STRING_LIBCALL(int,
+                       memcmp,
+                       const void *str1,
+                       const void *str2,
+                       size_t      count)
+DECLARE_STRING_LIBCALL(int,
+                       memcmp,
+                       const void *str1,
+                       const void *str2,
+                       size_t      count)
+DECLARE_STRING_LIBCALL(void *, memcpy, void *dest, const void *src, size_t n)
+DECLARE_STRING_LIBCALL(void *, memset, void *, int, size_t)
+DECLARE_STRING_LIBCALL(void *, memmove, void *dest, const void *src, size_t n)
+DECLARE_STRING_LIBCALL(void *, memchr, const void *, int, size_t)
+DECLARE_STRING_LIBCALL(void *, memrchr, const void *, int, size_t)
+DECLARE_STRING_LIBCALL(size_t, strlen, const char *str)
+DECLARE_STRING_LIBCALL(int, strncmp, const char *s1, const char *s2, size_t n)
+DECLARE_STRING_LIBCALL(char *, strncpy, char *dest, const char *src, size_t n)
+DECLARE_STRING_LIBCALL(int, strcmp, const char *s1, const char *s2)
+DECLARE_STRING_LIBCALL(char *,
+                       strnstr,
+                       const char *haystack,
+                       const char *needle,
+                       size_t      haystackLength)
+DECLARE_STRING_LIBCALL(char *, strchr, const char *s, int c)
+DECLARE_STRING_LIBCALL(size_t, strlcpy, char *dest, const char *src, size_t n)
+
+#undef DECLARE_STRING_LIBCALL
```

Comment:
If you add a CHERIOT prefix on this, we can move it into cdefs.h.  There's a similar macro in lib/atomic and it would be nice to refactor that to use this one at the same time.

Reply:
Done